lib/scholar/metrics/classification.ex

```diff
@@ -672,11 +674,70 @@ defmodule Scholar.Metrics.Classification do
         {true_positive, false_positive, false_negative}
 
       _ ->
-      {true_positive, false_positive, false_negative}
+        {true_positive, false_positive, false_negative}
     end
   end
 
-  defnp precision_recall_fscore_n(y_true, y_pred, beta, num_classes, average) do
+  @doc """
+  Calculates precision, recall, F-score and support for each
+  class. It also supports a `beta` argument which weights
+  recall more than precision by it's value.
+
+  ## Options
+
+  #{NimbleOptions.docs(@precision_recall_fscore_support_schema)}
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([0, 1, 1, 1, 1, 0, 2, 1, 0, 1], type: :u32)
+      iex> y_pred = Nx.tensor([0, 2, 1, 1, 2, 2, 2, 0, 0, 1], type: :u32)
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3)
+      #Nx.Tensor<
+        f32[3]
+        [0.6666666865348816, 0.6666666865348816, 0.4000000059604645]
+      >
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3, average: :macro)
+      #Nx.Tensor<
+        f32
+        0.5777778029441833
+      >
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3, average: :weighted)
+      #Nx.Tensor<
+        f32
+        0.6399999856948853
+      >
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3, average: :micro)
+      #Nx.Tensor<
+        f32
+        0.6000000238418579
+      >
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(Nx.tensor([1, 0, 1, 0]), Nx.tensor([0, 1, 0, 1]), 2, num_classes: 2, average: :none)
+      #Nx.Tensor<
+        f32[2]
+        [0.0, 0.0]
+      >
+  """
+  deftransform precision_recall_fscore_support(y_true, y_pred, beta \\ 1, opts \\ [])
+  
+  deftransform precision_recall_fscore_support(y_true, y_pred, beta, _opts) when is_list(beta) do
+    precision_recall_fscore_support(y_true, y_pred, 1, beta)
+  end
+
+  deftransform precision_recall_fscore_support(y_true, y_pred, beta, opts) when not is_list(beta) and is_list(opts) do
+    opts = NimbleOptions.validate!(opts, @precision_recall_fscore_support_schema)
+
+    precision_recall_fscore_support_n(
+      y_true,
+      y_pred,
+      beta,
+      opts[:num_classes],
+      opts[:average]
```

Comment:
Passing the atom `:none` directly instead of the `opts` list fails with:
```
** (Protocol.UndefinedError) protocol Nx.LazyContainer not implemented for :none of type Atom. This protocol is implemented for the following type(s): Any, Atom, Complex, Float, Integer, List, Map, Nx.Batch, Nx.Tensor, Tuple
```

Should the list be passed instead of single values?

Reply:
if you want to pass any atoms to the `defn` you should do this via `opts`. So yes, use keyword list instead :)